### ISSUE TITLE
querybuilder().then and .catch typings don't seem to exist

### ISSUE THREAD

**Issue by FabianTe** [tgriesser/knex#3331]
# Environment

**TypeScript Project**
Knex version: 0.18.3
Database + version: MSSQL 2012
OS: Windows 10

# Bug

When calling `.catch` or `.then` on an instance returned by `knex.queryBuilder()` the typescript compiler fails with this error:

![grafik](https://user-images.githubusercontent.com/7669818/60720037-c0491000-9f29-11e9-80e7-ca8779fcb1ec.png)

This does in fact work but the types seem to be off. I would like to not have to do an ugly type cast.

**Comment by lorefnon** [comment]
Can you please let us know the complete type of `query` ?

**Comment by lorefnon** [comment]
If your usage actually works, but the issue is only with typings, then most likely you are missing "es6" in `compilerOptions.lib` array in `tsconfig.json`. 

Without this typescript wouldn't be aware of the globals introduced in ES6 spec, including Promise. 

This may have worked for you in past because we used bluebird, which would provide its own promise compatible API. But we recently started using native promises so you will need to ensure the above.
